### PR TITLE
Add unit tests for xds package

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Adds additional tests to the xds package

# Context

https://github.com/solo-io/solo-projects/issues/5113

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/main/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/5113

### CODE REVIEW

File: projects/gloo/pkg/xds/cache_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/solo-io/gloo/projects/gloo/pkg/xds"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+var _ = Describe("Cache", func() {
+
+	It("NodeRoleHasher generates the correct ID", func() {
+		nodeRoleHasher := xds.NewNodeRoleHasher()
+		node := &envoy_config_core_v3.Node{}
+		Expect(nodeRoleHasher.ID(node)).To(Equal(xds.FallbackNodeCacheKey))
```
Comment (elcasteel) [inline projects/gloo/pkg/xds/cache_test.go:19]: nit: add a comment about what this is checking

**Review by elcasteel — COMMENTED** [review]
The new tests look good to me. I don't have enough context to know whether there are additional tests that we should add.

**Review by sam-heilbron — DISMISSED** [review]
LGTM

**Review by elcasteel — DISMISSED** [review]
for future reference github/bulldozer will automatically squash changes and it's more readable to leave the commit history intact until then

File: projects/gloo/pkg/xds/cache_test.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+var _ = Describe("Cache", func() {
+
+	It("NodeRoleHasher generates the correct ID", func() {
+		nodeRoleHasher := xds.NewNodeRoleHasher()
+		node := &envoy_config_core_v3.Node{}
+		// Ensure it returns the fallback key if the role field in the node metadata is not present
+		Expect(nodeRoleHasher.ID(node)).To(Equal(xds.FallbackNodeCacheKey))
```
Comment (sam-heilbron) [inline projects/gloo/pkg/xds/cache_test.go:20]: nit: I've tried to lean away from comments above assertions, and instead use the "optional desscription" (https://onsi.github.io/gomega/#annotating-assertions) field of an assertion to perform that documentation. That way if the assertion fails, it shows that relevant details.

Not a blocker, just wanted to share this pattern

**Review by sam-heilbron — APPROVED** [review]

**Review by elcasteel — APPROVED** [review]

File: projects/gloo/pkg/xds/cache_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+var _ = Describe("Cache", func() {
+
+	It("NodeRoleHasher generates the correct ID", func() {
+		nodeRoleHasher := xds.NewNodeRoleHasher()
+		node := &envoy_config_core_v3.Node{}
+		Expect(nodeRoleHasher.ID(node)).To(Equal(xds.FallbackNodeCacheKey),
+			fmt.Sprintf("Should return %s if the role field in the node metadata is not present", xds.FallbackNodeCacheKey))
```
Comment (sam-heilbron) [inline projects/gloo/pkg/xds/cache_test.go:20]: nit: https://onsi.github.io/gomega/#annotating-assertions you don't need to call `fmt.Sprintf`, that will be handled by Gomega directly. You can just pass the format and the parameters (example in the docs that I linked)

**Review by sam-heilbron — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Review by elcasteel — APPROVED** [review]

**Pull request merged**